flink-staging/flink-gelly/pom.xml

```diff
@@ -37,17 +37,17 @@ under the License.
 	<dependencies>
 		<dependency>
 			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-java</artifactId>
+			<artifactId>flink-java${scala.suffix}</artifactId>
 			<version>${project.version}</version>
 		</dependency>
 		<dependency>
 			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-clients</artifactId>
+			<artifactId>flink-clients${scala.suffix}</artifactId>
 			<version>${project.version}</version>
 		</dependency>
 		<dependency>
 			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-test-utils</artifactId>
+			<artifactId>flink-test-utils${scala.suffix}</artifactId>
 			<version>${project.version}</version>
 			<scope>test</scope>
 		</dependency>
```

Comment:
I can not comment below this line, but you forgot the `${scala.suffix}` for the `flink-optimizer`.